### ISSUE TITLE
[BUG] Inconsistency when it comes to the matplotlib requirement for autologging

### ISSUE THREAD

**Issue by Qfl3x** [mlflow/mlflow#6112]
### Willingness to contribute

Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.

### MLflow version

1.26.1

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 22.04 LTS
- **Python version**: 3.9.12
- **yarn version, if running the dev UI**: Not Applicable

### Describe the problem

There is inconsistency when it comes to the matplotlib dependency for the autologgers. 

If matplotlib is not present (common in case of running through a VM on the cloud or especially in Docker); the xgboost autologger signals the lack of matplotlib and continues with out. While the sklearn autologger crashes ungracefully.

### Code to reproduce issue

```python
import mlflow
mlflow.sklearn.autolog()

...
with mlflow.start_run():
    lr = LinearRegression()
    lr.fit(X_train, y_train)
```

### Other info / logs
For Sklearn:
```
Encountered unexpected error during sklearn autologging: No module named 'matplotlib'
```
For XGboost:
```
ERROR mlflow.xgboost: Failed to log feature importance plot. XGBoost autologging will ignore the failure and continue. Exception:
...
No module named `matplotlib`
```

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [X] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [X] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by harupy** [comment]
I think https://github.com/mlflow/mlflow/pull/5995 (merged after 1.26.1 release) fixed this issue.

**Comment by Qfl3x** [comment]
@harupy Works well!
